### PR TITLE
Refactor code for a little more speeeeeed :rocket: 

### PR DESCRIPTION
**Pull request by mhpob**
**BREAKING CHANGE**
Input data now needs to have dimensions of [FISH, TIME, RECEIVER] whereas it used to need [FISH, RECEIVER, TIME]

Code refactor:

- the structure is now "per individual" (loop) > "per time" (loop) > vectorized over receivers
  - this is why the input data went from [indiv, rec, time] to [indiv, time, rec]; apparently subsets happen more quickly from the higher indices to the lower indices
- `p0` and `sigma` are now generated quantities as we didn't need to use them as parameters (evaluate gradients, etc), just get the values. They are now just calculated at the end rather than every iteration
- distances have been moved to the model as to not save them
- we only ever use squared distances, so we don't take the sqrt of distance anymore
- similarly, we don't need to take the logit of everything as Stan has a `binomial_logit` function; we can just operate on the link scale and avoid running logits and inverse logits

Misc:

- Fixed tests accordingly
- nudge to v1.3.0
- require R 4.1.0 for base pipe
- fix documentation and update pkgdown

### CODE REVIEW

**Review by benjaminhlina — APPROVED** [review]

**Review by benjaminhlina — APPROVED** [review]

**Review by benjaminhlina — APPROVED** [review]

**Review by benjaminhlina — APPROVED** [review]

**Review by benjaminhlina — APPROVED** [review]

**Comment by benjaminhlina** [comment]
This looks good to go - I'll let you merge
